### PR TITLE
Measure status of control rods with folsom

### PR DESCRIPTION
**Pull request by ypaq**

### CODE REVIEW

File: bin/control_rods [hunk 1]
```diff
@@ -148,3 +165,11 @@ set_cookie(Node) ->
             erlang:set_cookie(Node, list_to_atom(Cookie))
     end.
 
+measure(CtrlRod, Value) ->
+    Node = node_name(),
+    Metric = metric(CtrlRod),
+    rpc:call(Node, folsom_metrics, notify, [Metric, Value, gauge]).
+
+metric(CtrlRod) ->
+    list_to_binary([<<"logplex.control_rod.">>, CtrlRod, <<".active">>]).
```
Comment (ricardochimal) [inline bin/control_rods]: `logplex.control-rod` ?  

ref: https://github.com/heroku/engineering-docs/blob/master/guides/metric-naming-conventions.md

**Comment by danp** [comment]
I don't quite follow this. Will this emit metrics at the time things are changed? What if logplex constantly emitted the values of these control rods?

File: bin/control_rods (hunk 1 above)
Comment (ypaq) [reply bin/control_rods]: Thanks for pointing this out. `shh` does the normalization [here](https://github.com/heroku/shh/blob/master/mm.go#L54-L60) for all metrics provided by folsom.

**Comment by ypaq** [comment]
@danp Yes, this will emit a metric when a control rod is changed.

**Comment by ypaq** [comment]
@danp We'd need a separate process that polls the control rods variables and emits the metrics.

**Comment by ypaq** [comment]
Now we measure the status of control rods via a new process `logplex_control_rods` that is supervised under the main logplex supervisor. This process triggers polling the state of each logplex control rod and emits a metric to folsom every minute.

File: src/logplex_control_rods.erl [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+     disable_api
+    ].
+
+notify_metric(CtrlRod) ->
+    Metric = metric(CtrlRod),
+    Value = status_to_value(CtrlRod),
+    folsom_metrics:notify(Metric, Value, gauge).
+
+metric(CtrlRod) ->
+    list_to_binary([<<"logplex.control_rod.">>, atom_to_list(CtrlRod), <<".active">>]).
```
Comment (danp) [inline src/logplex_control_rods.erl]: control_rod -> control-rod?

Comment (ypaq) [reply src/logplex_control_rods.erl]: `shh` normalizes the metrics (see @ricardochimal's comment above).

**Pull request merged**